### ISSUE TITLE
rsync erroneously reports success though failed

### ISSUE THREAD

**Issue by rabelux** [openmediavault/openmediavault#812]
# Description of issue/question
I have an rsync job that produced two mails on the last run.
first one saying:
```
ssh: connect to host servername port 22: Connection timed out
rsync: connection unexpectedly closed (0 bytes received so far) [sender] rsync error: unexplained error (code 255) at io.c(235) [sender=3.1.3]
```

with title `/var/lib/openmediavault/cron.d/rsync-9a66c88c-b8b6-4572-a142-fc58a65c2a1e | mail -E -s "Cron - Medien-Backup" -a`
Second mail:
```
Please wait, syncing </srv/dev-disk-by-label-xxx/> to <server:/share/medien/> ...
The synchronisation has completed successfully.
```

So I think there might be a bug with interpreting rsync output

### Versions report
5.5.9

**Comment by votdev** [comment]
> So I think there might be a bug with interpreting rsync output

OMV does not interpret any output, the email is submitted by the cron daemon when the job has any output to stdout.

You may check if the success email is not triggered by adding a `set -e` below the shebang line in `/var/lib/openmediavault/cron.d/rsync-9a66c88c-b8b6-4572-a142-fc58a65c2a1e` or by changing the shebang to `#!/bin/sh -e`.

**Comment by rabelux** [comment]
Added your suggested ` -e` and will see what happens next week.

Just ran it from webgui and received no mail but I guess no mails are sent when doing it that way?

 

I’d also suggest, if the effort isn’t too big, to call the hard coded `omv_log "\nThe synchronisation has completed successfully."`

in var/lib/openmediavault/cron.d/rsync only if actally succeded

 

Von: Volker Theile <notifications@github.com> 
Gesendet: Montag, 14. September 2020 19:13
An: openmediavault/openmediavault <openmediavault@noreply.github.com>
Cc: Alexander Raab <alexander.raab1@gmx.de>; Author <author@noreply.github.com>
Betreff: Re: [openmediavault/openmediavault] rsync erroneously reports success though failed (#812)

 

So I think there might be a bug with interpreting rsync output

OMV does not interpret any output, the email is submitted by the cron daemon when the job has any output to stdout.

You may check if the success email is not triggered by adding a set -e below the shebang line in /var/lib/openmediavault/cron.d/rsync-9a66c88c-b8b6-4572-a142-fc58a65c2a1e or by changing the shebang to #!/bin/sh -e.

**Comment by votdev** [comment]
> I’d also suggest, if the effort isn’t too big, to call the hard coded `omv_log "\nThe synchronisation has completed successfully."` in var/lib/openmediavault/cron.d/rsync only if actually succeded

The call is by intention where it is, so it does not make sense to relocate it.

**Comment by Martin-vH** [comment]
Hello folks, I'm having a similar issue using the USB Backup plug-in. The mail sent does always contain "The synchronisation has completed successfully." whereas several **"No space left on device"** logs are written to stderr and daemon.log. Shouldn't stderr be simply included in the mail? Additionally there should be some sort of failure indication, hence another mail perhaps?

**Comment by votdev** [comment]
Fixed in `openmediavault 5.5.14`.
